### PR TITLE
add gameflow docs

### PR DESCRIPTION
**Pull request by cykler**
#22 

Added the first draft of our `gameflow` document. I tried to answer as many of your questions as possible. But I guess we still have to refine quite a lot of it in person.

### CODE REVIEW

File: gameflow.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+## Game Logic
+
+The selected level starts and presents the user a random syllable (Hiragna / Katakana) as well as four possible answers.
+The available syllables depend on the selected level.
+
+If the correct answer was selected, the game immediately goes to the next syllable.
+
+If a wrong answer was selected, the game shows a quick indicator that something was wrong, removes a heart container and then goes to the next syllable.
+
+If the required amount of syllables have been answered, the game navigates to the success screen and informs the player how many Gems / Jewels he won as well as if he unlocked new things like syllables.
```
Comment (maxrimue) [inline gameflow.md]: Don't wanna be exclusive!
```suggestion
If the required amount of syllables have been answered, the game navigates to the success screen and informs the player how many Gems / Jewels they won as well as if they unlocked new things like syllables.
```

File: gameflow.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# Gameflow
+
+## Game Logic
+
+The selected level starts and presents the user a random syllable (Hiragna / Katakana) as well as four possible answers.
```
Comment (maxrimue) [inline gameflow.md]: Typo: "Hiragna"

Comment (maxrimue) [reply gameflow.md]: Also: by random you mean a randomly selected one within the level right? Maybe you could go even more specific here and mention at least an approximate number of syllables within a level, just to communicate the dimensions. The same goes for the segments down here as well.

File: gameflow.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+## Game Segments
+
+**Syllable**
+
+A single piece of gameplay in which the player has to select the correct counterpart for a given syllable in latin character.
+
+**Level**
+
+A collection of syllables the user has to complete. At the end of the level the user will be awarded in the form of gems.
```
Comment (maxrimue) [inline gameflow.md]: Note my point above concerning a rough estimate on how many Syllables we talk about here.

File: gameflow.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+**Gems / Jewels**
+
+Will be rewarded on completion of a level, no matter if the user succeeded or failed.
+Can be used to buy upgrades in the Gems / Jewels store. In this way the player can ease the growing complexitiy / difficulity of newly unlocked levels.
```
Comment (maxrimue) [inline gameflow.md]:
```suggestion
Can be used to buy upgrades in the gems / jewels store. This way, the player can ease the growing complexity / difficulty of newly unlocked levels.
```

File: gameflow.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+**Heart Container**
+
+Represents the amount of wrong answers a user is allowed to give per level. Will be refilled on start of a new level. Initially, 3 Heart Containers are given.
+
+**Upgrades**
+
+- Additional Heart Containers
+- New and more valubale Gems / Jewels
+- More time to give an answer
```
Comment (maxrimue) [inline gameflow.md]: I would perhaps rephrase this a bit for clarity, something like:
```suggestion
- Increased maximum time per syllable question
```

File: gameflow.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+Can be used to buy upgrades in the Gems / Jewels store. In this way the player can ease the growing complexitiy / difficulity of newly unlocked levels.
+
+**Heart Container**
+
+Represents the amount of wrong answers a user is allowed to give per level. Will be refilled on start of a new level. Initially, 3 Heart Containers are given.
+
+**Upgrades**
+
+- Additional Heart Containers
+- New and more valubale Gems / Jewels
```
Comment (maxrimue) [inline gameflow.md]:
```suggestion
- New and more valuable gems / jewels
```

Comment (maxrimue) [reply gameflow.md]: Do we perhaps want to add information here what this means? Like, what is the value of a Gem?
Also, shouldn't we settle for one name?

File: gameflow.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+Will be rewarded on completion of a level, no matter if the user succeeded or failed.
+Can be used to buy upgrades in the Gems / Jewels store. In this way the player can ease the growing complexitiy / difficulity of newly unlocked levels.
+
+**Heart Container**
+
+Represents the amount of wrong answers a user is allowed to give per level. Will be refilled on start of a new level. Initially, 3 Heart Containers are given.
+
+**Upgrades**
+
+- Additional Heart Containers
```
Comment (maxrimue) [inline gameflow.md]:
```suggestion
- Additional heart containers
```

File: gameflow.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+A collection of syllables the user has to complete. At the end of the level the user will be awarded in the form of gems.
+
+**Gems / Jewels**
+
+Will be rewarded on completion of a level, no matter if the user succeeded or failed.
+Can be used to buy upgrades in the Gems / Jewels store. In this way the player can ease the growing complexitiy / difficulity of newly unlocked levels.
+
+**Heart Container**
+
+Represents the amount of wrong answers a user is allowed to give per level. Will be refilled on start of a new level. Initially, 3 Heart Containers are given.
```
Comment (maxrimue) [inline gameflow.md]:
```suggestion
Represents the amount of wrong answers a user is allowed to give per level. Will be refilled on start of a new level. Initially, three heart containers are given.
```

File: gameflow.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+The selected level starts and presents the user a random syllable (Hiragna / Katakana) as well as four possible answers.
+The available syllables depend on the selected level.
+
+If the correct answer was selected, the game immediately goes to the next syllable.
+
+If a wrong answer was selected, the game shows a quick indicator that something was wrong, removes a heart container and then goes to the next syllable.
+
+If the required amount of syllables have been answered, the game navigates to the success screen and informs the player how many Gems / Jewels he won as well as if he unlocked new things like syllables.
+
+If the user runs out of Heart Containers the game navigates to the failure screen.
```
Comment (maxrimue) [inline gameflow.md]:
```suggestion
If the user runs out of heart containers the game navigates to the failure screen.
```

File: gameflow.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+## Game Segments
+
+**Syllable**
+
+A single piece of gameplay in which the player has to select the correct counterpart for a given syllable in latin character.
```
Comment (maxrimue) [inline gameflow.md]: Do all answers have to be correct to pass on?

File: gameflow.md [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+```
+{
+  /*
+   * Identifier of the current Level.
+   * Represents the actual level the user selected.
+   */
+  id: 1,
+
+  /*
+   * The amount of syllables the user has to answer
```
Comment (maxrimue) [inline gameflow.md]: "has to answer" - _correctly_?

Comment (cykler) [reply gameflow.md]: Not `correctly`. My idea was that the user has to answer a specific amount of syllables, doesn't matter if correct or not. He just has to make it through all the required syllables without consuming all of the heart containers.

File: gameflow.md (hunk 10 above)
Comment (cykler) [reply gameflow.md]: See comment below.

File: gameflow.md (hunk 6 above)
Comment (cykler) [reply gameflow.md]: Regarding value: We could, but this should be discussed first I think.
Regarding name: We should, but I figured this is worth discussing.

File: gameflow.md (hunk 2 above)
Comment (cykler) [reply gameflow.md]: But in the line below it's mentioned that the available syllables depend on the selected level. Doesn't that explain it already?

**Pull request merged**